modules/reporting/app/controllers/cost_reports_controller.rb

```diff
@@ -92,7 +92,13 @@ def index
     end unless performed?
   end
 
-  current_menu_item :index do |controller|
+  def show
+    query = CostQuery.find params[:id]
+    @subpage = query.name
+    super
+  end
```

Comment:
You are overriding the behaviour defined in https://github.com/opf/openproject/blob/dev/modules/reporting_engine/lib/report/controller.rb#L78-L86

Taking a look at the code there, I think that `@query` already contains the loaded query so you do not have to reload it from the database.

Reply:
> * If no cost query has been saved yet, the menu looks somewhat empty:

@ulferts Funny you are saying that... I changed that on purpose because I found the empty categories very confusing looking haha But I think you are right. If it's completely empty it looks somehow unnecessary and does not give a hint for what it is supposed to do... I will revert that change. Maybe a placeholder that says something like ' No private/public cost reports yet.' would be a good solution?

Reply:
> You are overriding the behaviour defined in https://github.com/opf/openproject/blob/dev/modules/reporting_engine/lib/report/controller.rb#L78-L86
> 
> Taking a look at the code there, I think that `@query` already contains the loaded query so you do not have to reload it from the database.

Okay, I will have a look at it